core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp

```diff
@@ -5174,10 +5152,13 @@ lower_select(JitCompContext *cc, x86::Assembler &a, COND_OP op, JitReg r0,
     CHECK_NCONST(r1);
     CHECK_KIND(r1, JIT_REG_KIND_I32);
 
-    if (r0 == r3 && r0 != r2) {
+    if (r0 == r3 && r0 != r2 && !cc->last_cmp_on_fp) {
```

Comment:
just curious, if the frontend follows the SSA rule, this branch will never be used, right? because `r0` will not be equal with `r3`.

Reply:
Yes, but it seems that the frontend doesn't follow the rule, the optimization was applied before, it happened when generating the IR for Jeff opcode.